### PR TITLE
Update logic for parsing sdk version number

### LINKED ISSUE #15909: flutter run command fails on Xcode 9.2
The [regular expression](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/ios/simulators.dart#L447) used to parse the sdk major version doesn't work on Xcode 9.2 when run with a simulator.  The `sdkNameAndVersion` is returned as "com.apple.CoreSimulator.SimRuntime.iOS-11-3", which doesn't match anything.

stack trace.
```
Unhandled exception:
NoSuchMethodError: The method 'group' was called on null.
Receiver: null
Tried calling: group(1)
#0      Object.noSuchMethod (dart:core-patch/dart:core/object_patch.dart:46)
#1      IOSSimulator.sdkMajorVersion (package:flutter_tools/src/ios/simulators.dart:449)
<asynchronous suspension>
#2      IOSSimulator.ensureLogsExists (package:flutter_tools/src/ios/simulators.dart:474)
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
I have a beta version of the iOS simulator at home.  It looks like the format of the sdk version has changed - see #15909 .

* Updates the parsing logic to handle `-` as well as ` `. 
* Adds a null aware operator.

Fixes #15909

### CODE REVIEW

File: packages/flutter_tools/test/ios/simulators_test.dart [hunk 1]
```diff
@@ -94,6 +94,20 @@ void main() {
     });
   });
 
+  group('sdkMajorVersion', () {
+    test('can be parsed from iOS-11-3', () async {
```
Comment (cbracken) [inline packages/flutter_tools/test/ios/simulators_test.dart]: Is this a new format starting with Xcode 9.3? 

Consider adding a comment here indicating the conditions under which we expect to see this (Xcode >= 9.3, I assume), as well as one on the test below -- if it's a version-specific behaviour, the previous behaviour covers at least Xcode 7 to 9.2.

Comment (jonahwilliams) [reply packages/flutter_tools/test/ios/simulators_test.dart]: I _think_ it is a beta version of the simulator. My xcode version was still 9.2 - I can verify this tonight

Comment (jonahwilliams) [reply packages/flutter_tools/test/ios/simulators_test.dart]: update - I have the exact same versions (9.2) on both my work and personal macs, down to the builder number. 

Likewise, both simulators are version 10.0.  However, my personal computer has a "beta" ribbon across it and slightly different version names.

new: SimulatorApp-850 CoreSimulator-518.16
old: SimulatorApp-835.5 CoreSimulator-494.33

Comment (jonahwilliams) [reply packages/flutter_tools/test/ios/simulators_test.dart]: added comment

**Comment by cbracken** [comment]
lgtm

**Review by cbracken — APPROVED** [review]

**Pull request merged**